storage/statedb/cache.go

```diff
@@ -1,8 +1,77 @@
 package statedb
 
+import "errors"
+
+type CacheType string
+
+const (
+	// Available stateDB cache types
+	LocalCache  CacheType = "LocalCache"
+	RemoteCache           = "RemoteCache"
+	HybridCache           = "HybridCache"
+)
+
+var (
+	errNotSupportedCacheType = errors.New("not supported stateDB Cache type")
+)
+
 // Cache interface the cache of stateDB
 type Cache interface {
 	Set(k, v []byte)
 	Get(k []byte) []byte
 	Has(k []byte) ([]byte, bool)
 }
+
+// NewCache creates one type of any supported stateDB caches.
+// TODO-Klaytn: refine input parameters after setting node flags
+func NewCache(cacheType CacheType, maxBytes int, redisEndpoint []string, redisCluster bool) (Cache, error) {
+	switch cacheType {
+	case LocalCache:
+		return NewFastCache(maxBytes), nil
+	case RemoteCache:
+		return NewRedisCache(redisEndpoint, redisCluster)
+	case HybridCache:
+		return NewHybridCache(maxBytes, redisEndpoint, redisCluster)
+	default:
+	}
+	return nil, errNotSupportedCacheType
+}
+
+func NewHybridCache(maxBytes int, endpoint []string, isCluster bool) (Cache, error) {
```

Comment:
```suggestion
func NewHybridCache(localCacheSizeBytes int, remoteCacheEndpoint []string, isCluster bool) (Cache, error) {
```

Reply:
I will refine input parameters later setting node flags. `redisCacheEndpoint` and `isCluster` may be placed in a struct.